### PR TITLE
Complete Authentication

### PR DESCRIPTION
**Pull request by nareshpoonia**
Complete Authentication

### CODE REVIEW

File: src/components/wishlist.js [hunk 1]
```diff
@@ -1,9 +1,95 @@
 import { useCartContext } from "../context/cart-context";
+import { useAuth } from "../context/auth-context";
+import { updateCart, addToCart, removeFromWishlist } from "../utilities";
+import { ACTION_TYPE } from "../utils/const";
 
 export function WishlistProducts(props) {
-  const { title, brand, price } = props.wishlist;
+  console.log(props);
```
Comment (omjadhav85) [inline src/components/wishlist.js]: remove this comment if not required

File: src/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      };
+      const response = await axios.post("/api/auth/login", guestLoginData);
+      localStorage.setItem("token", response.data.encodedToken);
+      setAuthToken(response.data.encodedToken);
+      const previousPath = location?.state?.from?.pathname || "*";
+      navigate(previousPath);
+    } catch (error) {
+      console.error(error);
+    }
+  }
```
Comment (omjadhav85) [inline src/context/auth-context.js:53]: i don't think a seperate function is needed for guest login, you can jus pass test credentials in login function itself

**Review by omjadhav85 — APPROVED** [review]
Few minor suggestions mentioned which can be implemented. Else, its good to merge! Great work!

File: src/components/product.js [hunk 3]
```diff
@@ -1,13 +1,110 @@
...
+            });
+          } catch (error) {
+            console.log(error);
+          }
+        })();
+      } else {
+        (async () => {
+          try {
+            const updatedCart = await addToCart(authToken, props.product);
+            console.log(updatedCart);
```
Comment (saadahmadhere) [inline src/components/product.js]: remove console.log

**Review by saadahmadhere — COMMENTED** [review]
-Clean code
-Good use of contextAPI for global state management.
Just a suggestion: Remove unwanted console.log()

File: src/components/product.js (hunk 3 above)
Comment (nareshpoonia) [reply src/components/product.js]: Thanks for pointing out, removed console.log

File: src/components/wishlist.js (hunk 1 above)
Comment (nareshpoonia) [reply src/components/wishlist.js]: Thanks for pointing out, removed console log

File: src/context/auth-context.js (hunk 2 above)
Comment (nareshpoonia) [reply src/context/auth-context.js:53]: Thanks for the comment, the function is there just for readability.

File: src/components/product.js (hunk 3 above)
Comment (nareshpoonia) [reply src/components/product.js]: Thanks for pointing out, removed console log

**Pull request merged**